source/test/F0.Minesweeper.Components.Tests/Logic/Cell/CellVisualisationManagerTests.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using F0.Minesweeper.Components.Abstractions.Enums;
+using F0.Minesweeper.Components.Logic.Cell;
+using FluentAssertions;
+using Xunit;
+
+namespace F0.Minesweeper.Components.Tests.Logic.Cell
+{
+	public class CellVisualisationManagerTests
+	{
+		[Fact]
+		public void GetVisualisation_UncoveredWithoutAdjacentMineCount_Throws()
+		{
+			// Arrange
+			CellVisualisationManager instanceUnderTest = new();
+
+			// Act && Assert
+			Func<CellVisualisation> methodUnderTest = () => instanceUnderTest.GetVisualisation(CellStatusType.Uncovered);
+			methodUnderTest.Should().Throw<ArgumentNullException>();
+		}
+
+		[Theory]
+		[MemberData(nameof(CellVisualisationManagerTestData.TestData), MemberType = typeof(CellVisualisationManagerTestData))]
+		public void GetVisualisation_CorrectParameters_ReturnsCorrectVisualisation(VisualisationData visualisationData)
+		{
+			// Arrange
+			CellVisualisationManager instanceUnderTest = new();
+
+			// Act 
+			CellVisualisation visualisation = instanceUnderTest.GetVisualisation(visualisationData.CellStatusType, visualisationData.AdjacentMineCount);
+
+			// Assert
+			visualisation.Content.Should().Be(visualisationData.ExpectedContent);
+			visualisation.CssClass.Should().NotBeNull();
+		}
+	}
+
+	public static class CellVisualisationManagerTestData
+	{
+		public static TheoryData<VisualisationData> TestData => GenerateTestData();
+
+		private static TheoryData<VisualisationData> GenerateTestData()
```

Comment:
We could have this _MemberData_ in the test-class directly, without a nested type, but I guess it's a matter of preference.

Reply:
I have actually no clue why it's not directly there. I personally prefer this too - unless this test data could be used in tests for other classes as well. And since it's used for just one test I'll move it!